plugins/chat/app/services/chat/auto_join_channel_batch.rb

```diff
@@ -0,0 +1,112 @@
+# NOTE: When changing auto-join logic, make sure to update the `settings.auto_join_users_info` translation as well.
+# frozen_string_literal: true
+
+module Chat
+  # Service responsible to create memberships for a channel and a section of user ids
+  #
+  # @example
+  #  Chat::AutoJoinChannelBatch.call(
+  #   chat_channel_id: 1,
+  #   starts_at: 27,
+  #   ends_at: 58,
+  #  )
+  #
+  class AutoJoinChannelBatch
+    include Service::Base
+
+    contract
+    model :channel
+    model :category
+    step :create_memberships
+    step :recalculate_user_count
+    step :publish_new_channel
+
+    class Contract
+      attribute :chat_channel_id, :integer
```

Comment:
Can we just change to `channel_id` now?

Reply:
We could probably do something like this while migrating:
```ruby
attribute :chat_channel_id, :integer
attribute :channel_id, :integer

validates :channel_id, presence: true

before_validation { self.channel_id ||= chat_channel_id }
```
And then we’ll just have to remove this extra piece of code later. Wdyt?

Reply:
Yes I just didn't want to add this on top of the rest, but we can do this :p